File: transektcount/src/main/java/com/wmstein/transektcount/widgets/EditTitleWidget.java
/*
 * Copyright (c) 2016. Wilhelm Stein, Bonn, Germany.
 */

package com.wmstein.transektcount.widgets;

import android.content.Context;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.wmstein.transektcount.R;

import java.util.Objects;

/*******************************************************
 * Used by EditSectionActivity and widget_edit_title.xml
 * Created by by milo on 05/05/2014.
 * Adopted for TransektCount by wmstein on 18.02.2016
 * Last edited on 2023-05-09
 */
public class EditTitleWidget extends LinearLayout
{
    final TextView widget_title;
    final EditText section_name;

    public EditTitleWidget(Context context, AttributeSet attrs)
    {
        super(context, attrs);
        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        Objects.requireNonNull(inflater).inflate(R.layout.widget_edit_title, this, true);
        widget_title = findViewById(R.id.widgeteditTitle);
        section_name = findViewById(R.id.editsectionName);
    }

    public void setWidgetTitle(String title)
    {
        widget_title.setText(title);
    }

    public void setSectionName(String name)
    {
        section_name.setText(name);
    }

    public String getSectionName()
    {
        return section_name.getText().toString();
    }

    public void setHint(String hint)
    {
        section_name.setHint(hint);
    }

}

File: transektcount/src/main/java/com/wmstein/transektcount/widgets/EditTitleWidget.kt
/*
 * Copyright (c) 2016. Wilhelm Stein, Bonn, Germany.
 */
package com.wmstein.transektcount.widgets

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.widget.EditText
import android.widget.LinearLayout
import android.widget.TextView
import com.wmstein.transektcount.R
import java.util.Objects

/*******************************************************
 * Used by EditSpeciesListActivity and widget_edit_title.xml
 * Created by by milo on 05/05/2014.
 * Adopted for TransektCount by wmstein on 18.02.2016,
 * last edited in Java on 2023-05-09,
 * converted to Kotlin on 2023-06-26,
 * Last edit on 2023-10-06.
 */
class EditTitleWidget(context: Context, attrs: AttributeSet?) : LinearLayout(context, attrs) {
    private val widgettitle: TextView
    private val sectionname: EditText

    init {
        val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        Objects.requireNonNull(inflater).inflate(R.layout.widget_edit_title, this, true)
        widgettitle = findViewById(R.id.widgeteditTitle)
        sectionname = findViewById(R.id.editsectionName)
    }

    fun setWidgetTitle(title: String?) {
        widgettitle.text = title
    }

    var sectionName: String?
        get() = sectionname.text.toString()
        set(name) {
            sectionname.setText(name)
        }

}